src/odemis/acq/stream_settings.py

```diff
@@ -0,0 +1,174 @@
+    """
+    Get the index of the config entry with the given name
+    :param config_data: list of config entries
+    :param name: name of config entry
+    :return: index of config entry with given name
+    """
+    try:
+        index = list(map(lambda x: x["name"], config_data)).index(name)
+    except:
+        index = None
+
+    return index
+
+
+class StreamSettings:
+    # TODO change the description
+    """
+    Read and write a list of most recently used (mru) settings from and to a JSON file respectively. The first element
+    in the list will be the latest used setting and the last element will be the oldest used setting. The list will
+    consist of finite number of entries which will update in the above mention order each time the list of settings
+    is updated.
+    """
+
+    def __init__(self, file_path, max_entries, config_data: List[Dict[str, Any]] = None):
+        """
+        :param file_path: full directory path to save the JSON file
+        :param max_entries: number of entries to save in the JSON file
+        :param config_data: list of stream controller settings where the first element is the latest
+         stream entry. The number of entries is defined by max_entries.
+        """
+        self.max_entries = max_entries
+        self.config_data = config_data
+        self.file_path = file_path
+        self._read()  # update the config_data by reading the JSON file
+        self.settings_order = None  # the ordered list of setting parameters used for loading any given stream
+
+    def _read(self):
+        logging.debug("Reading the most recently used stream settings in JSON")
+        if os.path.exists(self.file_path):
+            try:
+                with open(self.file_path, 'r') as jsonfile:
+                    self.config_data = json.load(jsonfile)
+            except json.decoder.JSONDecodeError as e:
+                logging.warning(f"Error decoding JSON: {e}")
+            except Exception as e:
+                logging.warning(f"Error in reading the stream settings: {e}")
+
+    def _write(self):
+        logging.debug("Writing the most recently used stream settings in JSON")
+        try:
+            if self.config_data:
+                with open(self.file_path, 'w') as jsonfile:
+                    json.dump(self.config_data, jsonfile, indent=2)
+        except Exception as e:
+            logging.warning(f"Error in writing the stream settings: {e}")
+
+    def update_data(self, new_entries: List[Dict[str, Any]]):
+        """
+        Update the saved stream setting with the list of new_entries. First element in JSON file
+        is the latest entry while the last element is the oldest stream setting. The file updates
+        its entries based on the new_entries while maintaining a fixed length of entries in the file.
+        :param new_entries: (list of stream controller settings) Each element in the list represent the setting
```

Comment:
"stream controller settings" is not correct -> drop the "type" as it's already defined in the signature.